### ISSUE TITLE
New rule: same line or newline for `@else` statements

### ISSUE THREAD

**Issue by ntwb** [kristerkari/stylelint-scss#82]
via https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#elseplacement

newline:
```css
@if {
  ...
}
@else {
  ...
}
```

same line:
```css
@if {
  ...
} @else {
  ...
}
```

**Comment by dryoma** [comment]
Wow, we still don't have it :) Thanks.

**Comment by kristerkari** [comment]
Yeah, definitely should have this.

**Comment by dryoma** [comment]
Hmm, seems this rule won't be that easy.

Firstly, to be able to control `@else` placement at all, a user would need to set up ignore options for [`block-closing-brace-newline-after`](http://stylelint.io/user-guide/rules/block-closing-brace-newline-after/) like so: `ignoreAtRules: ["if", "else"]`. I would allow this rule to take effect, but would lead to:

```scss
@if ($f == 1) {
 ... 
} @else {} a { display: block; }
```
(notice the `a { display: block; }`). To prevent that we'll have to control how the closing brace of the last `@else` in a row behaves. The options I see:
1. Somehow read the `block-closing-brace-newline-after`'s config if present and apply it, or
1. Use our own option, specific to this rule, something like
```json
"at-else-newline-before": [
  "always", {
    "lastClosingBraceNewlineAfter": "always"
  }
]
```
---------
Secondly, the space before `@else`. [`block-closing-brace-space-after`](http://stylelint.io/user-guide/rules/block-closing-brace-space-after/) doesn't give any viable option, so we'll have to control it. The question is, should it be an option to this rule, or a standalone rule? `at-else-space-before` soulds stylelint-ish, but the whole rule for a single statement... I don't know, doesn't sound well.

And all that said, should this rule still be `at-else-newline-before` or `at-else-behavior`?

Any thoughts?

**Comment by dryoma** [comment]
Hmm, seems this rule won't be that easy.

Firstly, to be able to control `@else` placement at all, a user would need to set up ignore options for [`block-closing-brace-newline-after`](http://stylelint.io/user-guide/rules/block-closing-brace-newline-after/) like so: `ignoreAtRules: ["if", "else"]`. I would allow this rule to take effect, but would lead to:

```scss
@if ($f == 1) {
 ... 
} @else {} a { display: block; }
```
(notice the `a { display: block; }`). To prevent that we'll have to control how the closing brace of the last `@else` in a row behaves. The options I see:
1. Somehow read the `block-closing-brace-newline-after`'s config if present and apply it, or
1. Use our own option, specific to this rule, something like
```json
"at-else-newline-before": [
  "always", {
    "lastClosingBraceNewlineAfter": "always"
  }
]
```
---------
Secondly, the space before `@else`. [`block-closing-brace-space-after`](http://stylelint.io/user-guide/rules/block-closing-brace-space-after/) doesn't give any viable option, so we'll have to control it. The question is, should it be an option to this rule, or a standalone rule? `at-else-space-before` soulds stylelint-ish, but the whole rule for a single statement... I don't know, doesn't sound well.

And all that said, should this rule still be `at-else-newline-before` or `at-else-behavior`?

Any thoughts? @kristerkari @jeddy3 @ntwb

**Comment by dryoma** [comment]
Yeah, so basically if a user needs standard behavior for if/else, they'll just stick to the core stylelint rules, otherwise theys specify them as ignored in `at-rule-empty-line-before` and `block-closing-brace-newline-after` and use our custom rules.

Ok then, let's rock. Thanks for the advice @jeddy3

**Comment by dryoma** [comment]
Will be published with the next minor.

**Comment by dryoma** [comment]
Available in 1.4.1
